client/blocks/gdpr-banner/index.jsx

```diff
@@ -29,83 +29,87 @@ const SIX_MONTHS = 6 * 30 * 24 * 60 * 60;
 
 const hasDocument = typeof document !== 'undefined';
 
-class GdprBanner extends Component {
-	static propTypes = {
-		recordCookieBannerOk: PropTypes.func,
-		recordCookieBannerView: PropTypes.func,
-		translate: PropTypes.func,
-	};
+function shouldShowBanner() {
+	// Don't render banner in SSR.
+	if ( ! hasDocument ) {
+		return false;
+	}
+	const cookies = cookie.parse( document.cookie );
+	if ( cookies.sensitive_pixel_option === 'yes' || cookies.sensitive_pixel_option === 'no' ) {
+		return false;
+	}
+	if ( isWpMobileApp() ) {
+		return false;
+	}
+	if ( isCurrentUserMaybeInGdprZone() ) {
+		return true;
+	}
+	return false;
+}
 
-	static defaultProps = {
-		recordCookieBannerOk: noop,
-		recordCookieBannerView: noop,
-		translate: identity,
-	};
+function GdprBanner( props ) {
+	const [ hideBanner, setHideBanner ] = useState( false );
+	const [ renderBanner, setRenderBanner ] = useState( false );
+	const translate = useTranslate();
 
-	state = {
-		showBanner: true,
-	};
+	const { recordCookieBannerOk, recordCookieBannerView } = props;
 
-	acknowledgeClicked = () => {
+	const acknowledgeClicked = useCallback( () => {
```

Comment:
Does `acknowledgeClicked` need to be memoized with `useCallback`? It's used only as a `<Button onClick>` prop, isn't it? The hook approach is to just not worry about creating a different handler on every render.

Reply:
It's still creating a new handler on every render, though, and either setting/removing events from the DOM, or doing something similar inside React's own event system. The question is mostly moot, since I suspect this component doesn't render terribly often, but it still seems to me that the general approach should be to memoise the handler, regardless of whether you're passing it to a component or a native element, right?

Reply:
Even if the target component is a native element, it won't lead to `domEl.removeEventListener( 'click', oldHandler )` and `domEl.addEventListener( 'click', newHandler )` on every render. React maintains only one DOM `click` listener on the `document` and uses the event bubbling and the `event.target` property to dispatch the event to the right handlers.

With hooks, it's common to recreate the callbacks on every render. They should be memoized only when not memoizing has some bad consequences: defeating `shouldComponentUpdate` in the child component, having to really re-attach DOM listeners or re-create timeouts...

Reply:
Done.